src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -135,9 +138,10 @@ public void delete(Query query, Class<?> clazz, IndexCoordinates index) {
 		return doMultiSearch(queries, classes, request);
 	}
 
-	private <T> List<Page<T>> doMultiSearch(List<? extends Query> queries, Class<T> clazz, MultiSearchRequest request) {
+	private <T> List<Page<SearchHit<T>>> doMultiSearch(List<? extends Query> queries, Class<T> clazz,
+			MultiSearchRequest request) {
 		MultiSearchResponse.Item[] items = getMultiSearchResult(request);
-		List<Page<T>> res = new ArrayList<>(queries.size());
+		List<Page<SearchHit<T>>> res = new ArrayList<>(queries.size());
```

Comment:
How about a new ticket that introduces a `MultiResult<T>` type? `List<Page<SearchHit<T>>>` is somewhat hard to read.